test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresUnreferencedCodeCapability.cs

```diff
@@ -163,14 +167,84 @@ static void RequiresUnreferencedCodeOnlyThroughReflection ()
 		{
 		}
 
-		// issue: https://github.com/mono/linker/issues/1607
-		// Linker currently doesn't report a warning in this case
-		// [ExpectedWarning ("IL2026", "--RequiresUnreferencedCodeOnlyThroughReflection--")]
+		[ExpectedWarning ("IL2026", "--RequiresUnreferencedCodeOnlyThroughReflection--")]
 		static void TestRequiresUnreferencedCodeOnlyThroughReflection ()
 		{
-			typeof (RequiresUnreferencedCodeAttribute)
+			typeof (RequiresUnreferencedCodeCapability)
 				.GetMethod (nameof (RequiresUnreferencedCodeOnlyThroughReflection), System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic)
 				.Invoke (null, new object[0]);
 		}
+
+		class BaseType
+		{
+			[RequiresUnreferencedCode ("Message for --VirtualMethodRequiresUnreferencedCode--")]
+			public virtual void VirtualMethodRequiresUnreferencedCode ()
+			{
+			}
+		}
+
+		class TypeWhichOverridesMethod : BaseType
+		{
+			[ExpectedWarning ("IL2046", "All overridden methods must have 'RequiresUnreferencedCodeAttribute'.")]
```

Comment:
I don't think testing the behavior with only partially annotated hierarchies is necessary. I think it's more important to test the case where the hierarchy is fully annotated (otherwise the user will get warnings anyway).